lib/jwt/jwk/key_base.rb

```diff
@@ -15,14 +15,19 @@ def initialize(options)
           options = { kid: options }
         end
 
-        @kid           = options[:kid]
+        @common_parameters = options[:common_parameters] || {}
+        @common_parameters = @common_parameters.transform_keys(&:to_sym) # Uniform interface
+        @common_parameters[:kid] = options[:kid] if options[:kid] # kid can be specified outside common_parameters
+
         @kid_generator = options[:kid_generator] || ::JWT.configuration.jwk.kid_generator
       end
 
       def kid
-        @kid ||= generate_kid
+        @common_parameters[:kid] ||= generate_kid
       end
 
+      attr_accessor :common_parameters
```

Comment:
This is public just because of the tests? I guess if we go with the suggested key accessor approach we can make this one private?